samza-yarn/src/main/java/org/apache/samza/job/yarn/YarnClusterResourceManager.java

```diff
@@ -262,7 +280,7 @@ public void releaseResources(SamzaResource resource) {
    */
 
   @Override
-  public void launchStreamProcessor(SamzaResource resource, CommandBuilder builder) throws SamzaContainerLaunchException {
+  public void launchStreamProcessor(SamzaResource resource, CommandBuilder builder)  {
```

Comment:
Need to poke around a bit more, but my gut feel is that we need a try-catch for this entire method, or even higher in the call stack to ensure that any exception in container launch causes onStreamProcessorLaunchFailure(). Also documentation on the abstract method javadoc, that no exceptions should be thrown in implementations of this method and instead the failure callback should be called. 

OR

AbstractContainerAllocator should catch all (client side) exceptions and invoke the failure callback directly.

Reply:
Good point. The current patch catches all sources of exceptions in the places they occur, and invokes the callbacks (wherever the sync-client threw a `SamzaContainerLaunchException`).

But, you're right - We need to be defensive, and I'd be +1 to adding an uber try-catch for safety. I'll also update the java-docs.